pallets/living-assets-evolution/src/lib.rs

```diff
@@ -34,36 +34,77 @@ pub mod pallet {
 	pub trait Config: frame_system::Config {
 		/// Because this pallet emits events, it depends on the runtime's definition of an event.
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// Limit for the length of `token_uri`
+		#[pallet::constant]
+		type MaxTokenUriLength: Get<u32>;
 		/// Type representing the weight of this pallet
 		type WeightInfo: WeightInfo;
+		/// Converts [`AccountId`] to [`H160`]
+		type AccountIdToH160: Convert<AccountIdOf<Self>, H160>;
 	}
 
 	/// Collection counter
 	#[pallet::storage]
 	#[pallet::getter(fn collection_counter)]
 	pub(super) type CollectionCounter<T: Config> = StorageValue<_, CollectionId, ValueQuery>;
 
-	// storage for the ownership of collections
+	/// Storage for the ownership of collections
 	#[pallet::storage]
 	#[pallet::getter(fn collection_owner)]
 	pub type CollectionOwner<T: Config> =
-		StorageMap<_, Blake2_128Concat, CollectionId, T::AccountId, OptionQuery>;
+		StorageMap<_, Blake2_128Concat, CollectionId, AccountIdOf<T>, OptionQuery>;
 
-	// Pallets use events to inform users when important changes are made.
-	// https://docs.substrate.io/main-docs/build/events-errors/
+	/// Minted assets
+	#[pallet::storage]
+	#[pallet::getter(fn asset_owner)]
+	pub type AssetOwner<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		CollectionId,
+		Blake2_128Concat,
+		Slot,
+		AccountIdOf<T>,
+		OptionQuery,
+	>;
+
+	/// Token URI which can override the default URI scheme and set explicitly
+	/// This will contain external URI in a raw form
+	#[pallet::storage]
+	#[pallet::getter(fn explicit_token_uri)]
+	pub type ExplicitTokenURI<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		CollectionId,
```

Comment:
The specification states _I can query tokenURI( collectionId, tokenId ) and it matches the provided value_, if I'm not wrong this query has the format: `tokenURI( collectionId, slot )`. It might be the specification is wrong?